src/library-authoring/components/ContainerCard.tsx

```diff
@@ -16,14 +17,73 @@ import { useSidebarContext } from '../common/context/SidebarContext';
 import { useLibraryRoutes } from '../routes';
 import BaseCard from './BaseCard';
 import messages from './messages';
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { ToastContext } from '../../generic/toast-context';
+import { useDeleteContainer, useRestoreContainer } from '../data/apiHooks';
 
 type ContainerMenuProps = {
   hit: ContainerHit,
 };
 
 const ContainerMenu = ({ hit } : ContainerMenuProps) => {
   const intl = useIntl();
-  const { contextKey, blockId } = hit;
+  const {
+    contextKey,
+    blockId,
+    usageKey: containerKey,
+    blockType: componentType,
+    displayName,
+  } = hit;
+  const {
+    sidebarComponentInfo,
+    closeLibrarySidebar,
+  } = useSidebarContext();
+
+  const [isConfirmingDelete, confirmDelete, cancelDelete] = useToggle(false);
+  const { showToast } = useContext(ToastContext);
+  const deleteContainerMutation = useDeleteContainer(containerKey);
+
+  let deleteWarningTitle;
+  let deleteText;
+  let deleteSuccess;
+  let deleteError;
+  if (componentType === 'unit') {
+    deleteWarningTitle = intl.formatMessage(messages.deleteUnitWarningTitle);
+    deleteText = intl.formatMessage(messages.deleteUnitConfirm, {
+      componentName: displayName,
+    });
+    deleteSuccess = intl.formatMessage(messages.deleteUnitSuccess);
+    deleteError = intl.formatMessage(messages.deleteUnitFailed);
+  }
```

Comment:
nit: It's going to be a little while before we support other component types.. I think it's ok to just let these messages be assigned as `const`s, with a "TODO" for later? We have the same in the backend.

```suggestion
  // TODO: support other container types besides 'unit'
  const deleteWarningTitle = intl.formatMessage(messages.deleteUnitWarningTitle);
  const deleteText deleteText = intl.formatMessage(messages.deleteUnitConfirm, {
    componentName: displayName,
  });
  const deleteSuccess = intl.formatMessage(messages.deleteUnitSuccess);
  const deleteError = intl.formatMessage(messages.deleteUnitFailed);
```

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/1773/commits/a17967fc25c7a1a8165188b29b31666cb4c76391